qiskit/circuit/gate.py

```diff
@@ -224,3 +226,18 @@ def broadcast_arguments(self, qargs: List, cargs: List) -> Tuple[List, List]:
             return Gate._broadcast_3_or_more_args(qargs)
         else:
             raise CircuitError('This gate cannot handle %i arguments' % len(qargs))
+
+    def validate_parameter(self, parameter):
+        """Gate parameters should be int, float, or ParameterExpression"""
+        if isinstance(parameter, (int, float, ParameterExpression)):
+            return parameter
+        elif isinstance(parameter, (np.integer, np.floating)):
+            return parameter.item()
+        elif isinstance(parameter, np.ndarray):
+            warn("Gate param type %s is being deprecated. Considering creating your own Gate "
+                 "subclass with the method validate_parameter to allow this param"
+                 " type." % type(parameter), DeprecationWarning, 3)
```

Comment:
Could you add a deprecation date so we know when the deprecation period has ended? 🙂

Reply:
Please don't put dates in the warning message. The deprecation timer starts at the release date which can be pretty far into the future and the deprecation is not for a fixed period of time, it could be 3 months but that's the minimum it could be a lot longer too. The deprecation policy has some suggestions on warning text if you want to give more of a feeling about when something will be removed: https://qiskit.org/documentation/contributing_to_qiskit.html#deprecation-warnings

Reply:
Ah, that's what I meant "3 months after the 0.16.0 release" -- sorry "date" wasn't the right word